src/Rules/Functions/StrvalFamilyParametersRule.php

```diff
@@ -0,0 +1,100 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Rules\RuleLevelHelper;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\Type;
+use PHPStan\Type\VerbosityLevel;
+use function in_array;
+use function sprintf;
+
+/**
+ * @implements Rule<Node\Expr\FuncCall>
+ */
+class StrvalFamilyParametersRule implements Rule
+{
+
+	private const FUNCTIONS = [
+		'strval',
+		'intval',
+		'floatval',
+		'doubleval',
+	];
+
+	public function __construct(
+		private RuleLevelHelper $ruleLevelHelper,
+	)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return Node\Expr\FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!($node->name instanceof Node\Name)) {
+			return [];
+		}
+		$name = $node->name->toString();
```

Comment:
Is this actually still necessary? I did a quick check and it appears to me, that this upper case function calls are already cast to lower case somewhere earlier.

Reply:
just add a unit test like `echo strVal($array) . PHP_EOL;` or `echo STRVAL($array) . PHP_EOL;` and you will see

Reply:
I could have sworn I did that and `$name` still was in all lowercase. But I just checked again and you're right it isn't. Well I already added it just in case, so this should be fine.